src/itest/resources/features/company-profile-error-retry.feature

```diff
@@ -0,0 +1,48 @@
+Feature: Error and retry scenarios for company profile
+
+  Scenario Outline: Processing company profile information unsuccessfully
+
+    Given Company profile api service is running
+    When I send PATCH request with raw payload "<data>" and company number "<company_number>"
+    Then I should receive <response_code> status code
+    And the CHS Kafka API is not invoked
+    And nothing is persisted in the database
+
+    Examples:
+      | data                               | company_number     | response_code |
+      | bad_request_payload                | 11748564           | 400           |
+
+  Scenario Outline: Retrieve company profile unsuccessfully as the company number does not exist
+
+    Given Company profile api service is running
+    When I send GET request with company number "<data>"
+    Then I should receive 404 status code
+
+    Examples:
+      | data     |
+      | 00006400 |
+
+  Scenario Outline: Processing company profile information while database is down
+
+    Given Company profile api service is running
+    And the company profile database is down
+    When I send PATCH request with payload "<data>" and company number "<data>"
+    Then I should receive 500 status code
+    And the CHS Kafka API is not invoked
+
+    Examples:
+      | data     |
+      | 11748564 |
+
+  Scenario Outline: Processing company profile information unsuccessfully but saves to database
+
```

Comment:
The scenario below is missing.

Scenario 5 (503 - Service Unavailable):
Given Company Profile API component is successfully running but MongoDB is not reachable
When ‘GET’ Rest endpoint is invoked with a valid company number but Repository throws an error (replicate MongoDB not reachable). NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 503 (Service Unavailable) to the client

Reply:
Changes done